### ISSUE TITLE
Evaluation stuck indefinitely on large package set

### ISSUE THREAD

**Issue by DavHau** [hercules-ci/support#67]
[This job](https://hercules-ci.com/github/nix-community/dream2nix-nodejs-auto/jobs/45), which is a flake exporting about 5000 packages, is stuck indefinitely.

There is an `Evaluation limit reached` error, but the evaluation does not terminate.

The repo is automatically updated by a bot, so the number of such stuck hercules jobs increases every day:
https://github.com/nix-community/dream2nix-nodejs-auto/commits/data
I will now cancel the jobs manually except the one which I linked above, to allow for debugging.

**Comment by roberth** [comment]
I've made a change to the backend that should help with stuckness, though you'll have to cancel the (second-to) last job by hand.

> `Evaluation limit reached` error,

This is actually due to the large amount of derivations rather than the intended limit on attributes. `hercules-ci-agent` `master` should fix this. I've created a PR to update it directly, https://github.com/nix-community/infra/pull/261. Otherwise it will take an agent release, a nixpkgs haskell-updates update, and channel update for this fix to reach you.
